client/app/bundles/HelloWorld/containers/HelloWorld.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { PropTypes } from 'react';
+import HelloWorldWidget from '../components/HelloWorldWidget';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import Immutable from 'immutable';
+import * as helloWorldActionCreators from '../actions/helloWorldActionCreators';
+
+function select(state) {
+  // Which part of the Redux global state does our component want to receive as props?
+  // Note the use of `$$` to prefix the property name because the value is of type Immutable.js
+  return { $$helloWorldStore: state.$$helloWorldStore };
+}
+
+// Simple example of a React "smart" component
+const HelloWorld = (props) => {
+  const { dispatch, $$helloWorldStore } = props;
+  const actions = bindActionCreators(helloWorldActionCreators, dispatch);
+  const { updateName } = actions;
+  const name = $$helloWorldStore.get('name');
+
+  // This uses the ES2015 spread operator to pass properties as it is more DRY
+  // This is equivalent to:
+  // <HelloWorldWidget $$helloWorldStore={$$helloWorldStore} actions={actions} />
+  return (
+    <HelloWorldWidget {...{ updateName, name }} />
+  );
+};
```

Comment:
ive seen more elegant ways to create smart components where you can map state to props and dispatcher to props.